nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java

```diff
@@ -121,18 +146,47 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             if (createContainer && !containerClient.exists()) {
                 containerClient.create();
             }
-            BlobClient blobClient = containerClient.getBlobClient(blobName);
 
+            BlobClient blobClient = containerClient.getBlobClient(blobName);
+            final BlobRequestConditions blobRequestConditions = new BlobRequestConditions();
+            Map<String, String> attributes = new HashMap<>();
+            attributes.put(ATTR_NAME_CONTAINER, containerName);
+            attributes.put(ATTR_NAME_BLOBNAME, blobName);
+            attributes.put(ATTR_NAME_PRIMARY_URI, blobClient.getBlobUrl());
+            attributes.put(ATTR_NAME_LANG, null);
+            attributes.put(ATTR_NAME_MIME_TYPE, "application/octet-stream");
             long length = flowFile.getSize();
 
-            try (InputStream rawIn = session.read(flowFile);
-                 BufferedInputStream bufferedIn = new BufferedInputStream(rawIn)) {
-                blobClient.upload(bufferedIn, length);
+            try {
+                if (conflictResolution != AzureStorageConflictResolutionStrategy.REPLACE_RESOLUTION) {
+                    blobRequestConditions.setIfNoneMatch("*");
+                }
+
+                try (InputStream rawIn = session.read(flowFile)) {
+                    final BlobParallelUploadOptions blobParallelUploadOptions = new BlobParallelUploadOptions(toFluxByteBuffer(rawIn));
+                    blobParallelUploadOptions.setRequestConditions(blobRequestConditions);
+                    Response<BlockBlobItem> response = blobClient.uploadWithResponse(blobParallelUploadOptions, null, Context.NONE);
+                    BlockBlobItem blob = response.getValue();
+                    attributes.put(ATTR_NAME_BLOBTYPE, BlobType.BLOCK_BLOB.toString());
+                    attributes.put(ATTR_NAME_ETAG, blob.getETag());
+                    attributes.put(ATTR_NAME_LENGTH, String.valueOf(length));
+                    attributes.put(ATTR_NAME_TIMESTAMP, String.valueOf(blob.getLastModified()));
+                }
+            } catch (BlobStorageException e) {
+                if (conflictResolution == AzureStorageConflictResolutionStrategy.FAIL_RESOLUTION) {
+                    throw e;
+                }
+                final BlobErrorCode errorCode = e.getErrorCode();
+                if (errorCode == BlobErrorCode.BLOB_ALREADY_EXISTS) {
+                    getLogger().warn("Blob already exists: remote blob not modified. Transferring {} to success", flowFile);
```

Comment:
Since this is an expected scenario, I wouldn't log with a warning log level. I would instead use debug.

Reply:
I agree, but ADLS processor does the same thing. That doesn't mean of course that we shouldn't fix it.

But perhaps separately?

Reply:
I have changed it to "info" – most processors are configured for WARN and it is perhaps slightly more informational than debug level I think, especially since we have defined success as also including these ignored blobs.

Reply:
PutHDFS is also logging this case with info, so it is good. Thanks.